src/Mutation/MediaItemCreate.php

```diff
@@ -151,8 +151,26 @@ public static function mutate_and_get_payload() {
 			 * Set the file name, whether it's a local file or from a URL.
 			 * Then set the url for the uploaded file
 			 */
-			$file_name         = basename( $input['filePath'] );
-			$uploaded_file_url = $input['filePath'];
+			$file_name           = basename( $input['filePath'] );
+			$uploaded_file_url   = $input['filePath'];
+			$sanitized_file_path = sanitize_file_name( $input['filePath'] );
+
+			// Check that the filetype is allowed
+			$check_file = wp_check_filetype( $uploaded_file_url );
+
+			// if the file doesn't pass the check, throw an error
+			if ( ! $check_file['ext'] || ! $check_file['type'] || ! wp_http_validate_url( $uploaded_file_url ) ) {
+				throw new UserError( sprintf( __( 'Invalid filePath "%s"', 'wp-graphql' ), $input['filePath'] ) );
+			}
+
+			// prevent the filePath from being submitted with a non-allowed protocols
+			$allowed_protocols = [ 'https', 'http', 'file' ];
+			$allowed_protocols = apply_filters( 'graphql_media_item_create_allowed_protocols', $allowed_protocols );
```

Comment:
Would it make sense to initialize the current `$protocol` earlier and pass it through the filter?

Pros:
- make it easier to support more complex validation without a dedicated action/filter.

Cons:
- make it easier to ~support more complex validation~ _introduce unwanted edge cases_  without a dedicated action/filter

Reply:
yes, good call. Will also pass `$input`, `$context` and `$info`.  🙌🏻